File: apps/android/feature/canvas/domain/src/main/java/com/scribblefit/feature/canvas/domain/model/FeedItem.kt
package com.scribblefit.feature.canvas.domain.model

import com.scribblefit.feature.ai.domain.model.ParsedWorkout
import com.scribblefit.feature.ai.domain.model.SuggestionType

sealed class FeedItem {
    data class Prompt(
        val id: String,
        val timestamp: Long,
        val text: String,
        val emoji: String,
        val type: SuggestionType
    ) : FeedItem()

    data class Scribble(
        val id: String,
        val timestamp: Long,
        val rawText: String,
        val status: ScribbleStatus
    ) : FeedItem()

    data class Confirmation(
        val id: String,
        val timestamp: Long,
        val workout: ParsedWorkout,
        val scribbleId: String
    ) : FeedItem()

    data class Insight(
        val id: String,
        val timestamp: Long,
        val text: String,
        val emoji: String
    ) : FeedItem()
}

enum class ScribbleStatus { PENDING, PROCESSING, FAILED, COMPLETED }

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/Models/FeedItem.swift
import Foundation

/**
 * Represents an item in the Intelligent Canvas feed.
 */
public enum FeedItem: Sendable, Identifiable {
    case prompt(PromptItem)
    case scribble(ScribbleItem)
    case confirmation(ConfirmationItem)
    case insight(InsightItem)
    
    public var id: String {
        switch self {
        case .prompt(let item): item.id
        case .scribble(let item): item.id
        case .confirmation(let item): item.id
        case .insight(let item): item.id
        }
    }
    
    public var timestamp: Date {
        switch self {
        case .prompt(let item): item.timestamp
        case .scribble(let item): item.timestamp
        case .confirmation(let item): item.timestamp
        case .insight(let item): item.timestamp
        }
    }
}

public struct PromptItem: Sendable, Identifiable {
    public let id: String
    public let timestamp: Date
    public let text: String
    public let emoji: String
    public let type: SuggestionType
}

public struct ScribbleItem: Sendable, Identifiable {
    public let id: String
    public let timestamp: Date
    public let rawText: String
    public let status: ScribbleStatus
}

public struct ConfirmationItem: Sendable, Identifiable {
    public let id: String
    public let timestamp: Date
    public let workout: ParsedWorkout
    public let scribbleId: String
}

public struct InsightItem: Sendable, Identifiable {
    public let id: String
    public let timestamp: Date
    public let text: String
    public let emoji: String
}

public enum ScribbleStatus: String, Codable, Sendable {
    case pending, processing, failed, completed
}
